staging/src/k8s.io/client-go/dynamic/client_pool.go

```diff
@@ -70,6 +71,13 @@ func NewClientPool(config *restclient.Config, mapper meta.RESTMapper, apiPathRes
 	}
 }
 
+// Instantiates a new dynamic client pool with the given config.
+func NewDynamicClientPool(cfg *restclient.Config) ClientPool {
+	// TODO: should use a dynamic RESTMapper built from the discovery results.
+	restMapper := api.Registry.RESTMapper()
```

Comment:
There is also a registry for client-go. Note that there is a difference between client and kube types. Kube's api.Registry is relevant only on the apiserver side. If you talk to the apiserver via a client most probably you mean client-go's registry here.